libp2p/switch.nim

```diff
@@ -273,11 +278,13 @@ proc accept(s: Switch, transport: Transport) {.async.} = # noraises
     except CancelledError as exc:
       trace "releasing semaphore on cancellation"
       upgrades.release() # always release the slot
+    except TransportUseClosedError:
+      trace "Graceful shutdown in accept loop, exiting"
```

Comment:
we can use the same as below `"Exception in accept loop, exiting", exc = exc.msg`

Reply:
Don't you think that might trigger the same reactions, from users, that @cheatfate was mentioning in the issue?

Reply:
I think the issue was about the `error` log level. We can keep the one you used, my comment was about the rest.

Reply:
Users aren't supposed to use `trace` unless they are trying to debug an issue. That was precisely my intention suggesting `"Exception in accept loop, exiting", exc = exc.msg`, adding more useful info to the log.

Reply:
It doesn't need to be exactly this msg, but something more specific about what happened than the one currently used, plus the exception msg.

Reply:
Okay, let's see if I understood correctly:
1. When a `TransportUseClosedError` is raised, the whole server should stop?
2. Should the same behaviour happen with other `CatchableError`s as well?
3. What are you referring to with `acceptLoop` the indefinite calls to the `accept` proc? When reading your point 3 it doesn't seem that way, and it confuses me.

Reply:
> So to properly fix this issue you should first break/cancel/stop your loop which perform connection acceptance, and only after that you should close chronos server.

Switch accept loop uses the transport `running` flag, so it's possible that the current accept will run when the transport has been stopped and this will be raised https://github.com/vacp2p/nim-libp2p/blob/9059a8aced87741de98813bfe0ef2ed63e0217ab/libp2p/transports/tcptransport.nim#L203.

Reply:
@cheatfate

Reply:
The issue might be fixed by https://github.com/vacp2p/nim-libp2p/pull/1095#discussion_r1593835102. I'm tempted to close this PR.

Reply:
At the moment, I don't know how to reproduce it, if it hasn't been fixed yet.